lxd/storage.go

```diff
@@ -187,24 +200,20 @@ func storageForFilename(d *Daemon, filename string) (storage, error) {
 	config := make(map[string]interface{})
 	storageType := storageTypeDir
 
-	filesystem, err := filesystemDetect(filename)
-	if err != nil {
-		return nil, fmt.Errorf("couldn't detect filesystem for '%s': %v", filename, err)
-	}
-
-	lvLinkPath := filename + ".lv"
-	if shared.PathExists(lvLinkPath) {
+	if shared.PathExists(filename + ".lv") {
 		storageType = storageTypeLvm
-		lvPath, err := os.Readlink(lvLinkPath)
+		lvPath, err := os.Readlink(filename + ".lv")
 		if err != nil {
-			return nil, fmt.Errorf("couldn't read link dest for '%s': %v", lvLinkPath, err)
+			return nil, fmt.Errorf("couldn't read link dest for '%s': %v", filename+".lv", err)
 		}
 		vgname := filepath.Base(filepath.Dir(lvPath))
 		config["vgName"] = vgname
-
-	} else if filesystem == "btrfs" {
+	} else if shared.PathExists(filename + ".btrfs") {
 		storageType = storageTypeBtrfs
+	} else if shared.PathExists(filename + ".zfs") {
+		storageType = storageTypeZfs
```

Comment:
It's not missing. I don't have a way to know what the zfsPool is at this point and since we don't support having more than one pool, it's fine not to set it which will make Init pick up the one set in the server config.

I still support config["zfsPool"] in Init to make it easier to write tests for the Zfs driver as that allows bypassing the pool name set in the server config.